packages/nextjs/src/edge/transport.ts

```diff
@@ -8,6 +9,57 @@ export interface EdgeTransportOptions extends BaseTransportOptions {
   headers?: { [key: string]: string };
 }
 
+const DEFAULT_TRANSPORT_BUFFER_SIZE = 30;
+
+/**
+ * This is a modified promise buffer that collects tasks until drain is called.
+ * We need this in the edge runtime because edge function invocations may not share I/O objects, like fetch requests
+ * and responses, and the normal PromiseBuffer inherently buffers stuff inbetween incoming requests.
+ */
+class IsolatedPromiseBuffer {
+  // We just have this field because the promise buffer interface requires it.
+  // If we ever remove it from the interface we should also remove it here.
+  public $: Array<PromiseLike<TransportMakeRequestResponse>> = [];
+
+  private _taskProducers: (() => PromiseLike<TransportMakeRequestResponse>)[] = [];
```

Comment:
They're of a different type. We need to store producers in this case because we do not want to actually create the promises right away. Producers are `() => Promise` whereas `$` is just an array of `Promises`.

Reply:
I see - so we are essentially making all outgoing fetch requests lazy, because we want to only send them when `drain` is called.

I guess this works for now, but there is always the risk that data does not get sent to Sentry because of this right? This is an important caveat that should be put in as a code comment, that we knew about this tradeoff but we still proceeded with this.

Reply:
Generally, we flush on every edge function invocation so we should be sending everything. This implementation has the drawback though that there is a hard cap on the amount of events sent for any given edge function invocation: https://github.com/getsentry/sentry-javascript/pull/6980/files#diff-750cbb0cd6b1f70c9ca7a215d0e9ef1e0268b822bf194ccf49feddb7a6341d64R19-R20